gamemodes/terrortown/gamemode/shared/hud_elements/tttwswitch/old_ttt_wswitch.lua

```diff
@@ -117,61 +148,33 @@ if CLIENT then
 		return true
 	end
 
-	function HUDELEMENT:Initialize()
-		WSWITCH:UpdateWeaponCache()
-
-		self:SetBasePos(ScrW() - (width + self.margin * 2), ScrH() - self.margin)
-		self:SetSize(width, -height)
-
-		BaseClass.Initialize(self)
-
-		self.defaults.resizeableY = false
-		self.defaults.minHeight = height
-	end
-
-	function HUDELEMENT:PerformLayout()
-		local basepos = self:GetBasePos()
-
-		self:SetPos(basepos.x, basepos.y)
-		self:SetSize(width, -height)
-
-		BaseClass.PerformLayout(self)
+	function HUDELEMENT:ShouldDraw()
+		return HUDEditor.IsEditing or WSWITCH.Show
 	end
 
 	function HUDELEMENT:Draw()
-		if not WSWITCH.Show and not HUDManager.IsEditing then return end
-
-		local client = LocalPlayer()
+		local weaponList = {}
 		local weps = WSWITCH.WeaponCache
-		local count = #weps
-		local tmp = height + self.margin
-		local h = count * tmp
-		local basepos = self:GetBasePos()
-
-		self:SetPos(basepos.x, basepos.y)
-		self:SetSize(width, -h)
-
-		local pos = self:GetPos()
-		local x_elem = pos.x
-		local y_elem = pos.y
-		local col = self.col_dark
-
 		for k, wep in ipairs(weps) do
-			if WSWITCH.Selected == k then
-				col = self.col_active
-			else
-				col = self.col_dark
-			end
-
-			self:DrawBarBg(x_elem, y_elem, width, height, col)
+			table.insert(weaponList, {h = height})
```

Comment:
use table.Count() and multiply or just use "for k in ipairs(weps) do" (but seems to be odd)